draft-jmiller-json-proof-token.md

```diff
@@ -87,51 +87,43 @@ In order to generate a variety of ZKPs of knowledge, range, membeship, or other
 
 It is suggested that the claim names used with JPTs come from those in the IANA JSON Web Token Claims Registry, when those fit the application's needs.
 
-# Layout
+# Claims
 
-Using a JSON Proof Token requires combining information from two sources: the layout and the payloads. Given the design considerations, the simplest solution is to represent the token's encompassing JSON body in an external shared definition, while preserving only the individual claim values in the JWP container.
+Using a JSON Proof Token requires combining information from two sources: the claim names and the payloads.  The simplest solution is to list the claim names in an ordered array that aligns with the included payloads.  This claims array can be conveniently included in the JWP Protected Header using the `claims` key.
 
-The layout definition MUST be accessible to and known by all parties that will process the JPT.  Since this is also true of the public key, the ideal location for the layout definition is in the JWK itself. (Note: The layout definition could move to a standalone JSON Proof Key specification.)
+When the claims array is stored in the header, any variations of it are disclosed to the verifier and can be used to correlate and link usages.  Given the privacy design considerations around linkability it is recomended that the claims are defined external to an individual JPT and either referenced or known by the application context.
 
-A layout is the symbolic equivalent to the JSON body of a JWT, containing all of the claim names along with any additional structures or fixed metadata.  The individual claim values are not included in the layout and instead are replaced by payload references.
+In order to facilitate this external definition of the claim names, an additional `cid` key is defined with a required digest value calculated as defined here.  This `cid` can be used similar to a `kid` in order to ensure externally resolve and then verify that the correct list of claim names are being used when processing the payloads containing the claim values.
 
-These claim value references can be present in two different types: direct and indirect.  A direct reference is always an array with a single integer value, acting as an index into the array of payloads contained in the JPT.  All payloads referenced directly MUST be the base64url encoding of the UTF-8 representation of a JSON value.
+If there is an associated JWK containing the signing key information, the `claims` key is also registered there as a convenient location for the claim names.
```

Comment:
I am against this language... keys should not store claims, it violates both the familiarity goal of the spec.

Reply:
Also an example and normative language are needed here.